addon/components/frost-bunsen-input-select/component.js

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash'
+import Ember from 'ember'
+import computed, {readOnly} from 'ember-computed-decorators'
+
+import InputMixin from 'ember-frost-bunsen/mixins/input'
+import ListInputMixin from 'ember-frost-bunsen/mixins/list-input'
+import layout from './template'
+
+export default Ember.Component.extend(InputMixin, ListInputMixin, {
```

Comment:
I don't think this justifies the complexity. How about this alternative using composition:

```
import ListInput from 'ember-frost-bunsen/sub-components/list-input'
import layout from './template'

export default Ember.Component.extend(InputMixin, ListInputMixin, {
  classNames: [
    'frost-bunsen-input-select',
    'frost-field'
  ],
  layout,

  // ======================================================
  //                 List Input Interface
  // ======================================================
 
  init() {
    this._super(...arguments)
    this.set('options', new ListInput())
  }

  setOptions() {
    this.get('options').setOptions()
  }

  didReceiveAttrs () {
    this.super(...arguments)
    this.setOptions()
  }

  setEnumValues(values) {
    this.get('options').setEnumValues()
  }

  setAsyncDataValues (modelDef, queryDef) {
    this.get('options').setAsyncDataValues()
  }
  // ==========================================================================
  // Computed Properties
  // ==========================================================================

  @readOnly
  @computed('cellConfig.properties.type')
  inputType (type) {
    return type || 'text'
  },

 ```